client-side/src/components/settings/Settings.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { useTranslation } from 'react-i18next'
+import GenericButton  from '../../stories/Button/GenericButton.jsx';
+import GenericInput from '../../stories/GenericInput/genericInput.jsx'
+
+
+const Setting = (currentUser) => {
+   
+    const preferencesId=currentUser.currentUser.preferences._id;
```

Comment:
please add default value to the currentUser or defined it as isRequired,
also, why it's currentUser.currentUser?